packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -329,11 +329,17 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
           widget.expansionCallback!(childIndex, false);
         }
       }
+        //Because when _handlePressed the state is not yet expanded
+      widget.expansionCallback?.call(index, !isExpanded);
 
-      setState(() {
+       setState(() {
         _currentOpenPanel = isExpanded ? null : pressedChild;
       });
     }
+    else
+    {
+       widget.expansionCallback?.call(index, !isExpanded);
```

Comment:
Since it looks like this is called either way (in the if or else block), can the else block be removed and it just be called unconditionally after the if statement?

Reply:
Yes, that was an oversight. I refactored and all tests are still passing.